modules/apps/object/object-web/src/main/java/com/liferay/object/web/internal/object/entries/portlet/action/EditObjectEntryRelatedModelMVCActionCommand.java

```diff
@@ -86,6 +87,18 @@ private void _addObjectRelationshipMappingTableValues(
 
 				sendRedirect(actionRequest, actionResponse, redirect);
 			}
+			else if (exception instanceof
+						PrincipalException.MustHavePermission) {
+
+				SessionErrors.add(actionRequest, exception.getClass());
+
+				hideDefaultErrorMessage(actionRequest);
+
+				String redirect = ParamUtil.getString(
+					actionRequest, "redirect");
+
+				sendRedirect(actionRequest, actionResponse, redirect);
```

Comment:
Inline it please

Reply:
Done [here](https://github.com/liferay-bpm/liferay-portal/pull/4150/commits/5af100de3466677ff90b767c86fd7a7d35a998e3)